pkg/apis/extensions/validation/validation.go

```diff
@@ -346,15 +346,20 @@ func ValidateJobSpec(spec *extensions.JobSpec) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldInvalid("completions", spec.Completions, isNegativeErrorMsg))
 	}
 
-	selector := labels.Set(spec.Selector).AsSelector()
-	if selector.Empty() {
-		allErrs = append(allErrs, errs.NewFieldRequired("selector"))
+	if len(spec.Selector.MatchLabels)+len(spec.Selector.MatchExpressions) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("(matchLabels or matchExpressions)"))
 	}
-
+	allErrs = append(allErrs, ValidatePodSelector(spec.Selector).Prefix("selector")...)
+	selector, err := extensions.PodSelectorAsSelector(spec.Selector)
+	if err != nil {
+		return allErrs
+	}
+	//TODO(mikedanese): reexamine wrt selector
```

Comment:
Looks fine, though the field name isn't specified correctly to NewFieldInvalid (s/template.labels/template.metadata.labels/).